### ISSUE TITLE
Enhancement: interaction command to infer non-unique levels

### ISSUE THREAD

**Issue by identicalsnowflake** [agda/agda#2427]
```agda
record Functor (f : ∀ {ℓ} → Set ℓ → Set ℓ) (ℓ₁ ℓ₂ : Level) : Set _ where
  field
    fmap : ∀ {t₁ : Set ℓ₁} {t₂ : Set ℓ₂} → (t₁ → t₂) → f t₁ → f t₂
```

I think Agda is smarter than it realizes here: if you load the above, Agda will complain that it's not smart enough to figure out what level should go in `_`; but if you put the wrong answer (e.g., `ℓ₁`) in this position, suddenly Agda proceeds to tell you you're wrong and give the correct answer!

This results in an odd workflow where I deliberately put in wrong answers so that Agda will correct me with the correct answer that it claims it's not smart enough to deduce.

**Comment by jespercockx** [comment]
The reason for this behaviour is that Agda checks the record signature separately from the field types, so information about the fields can never flow back to the signature. Maybe Agda could leave the sort meta in the signature open until after the fields are added, allowing the `fitsIn` function from `TypeChecking.Rules.Data` to fill it in?

**Comment by andreasabel** [comment]
Actually, the `Level` meta `_` is not uniquely determined.  It is constrained (press `C-c C-=`), thus, you get an error if you put in a wrong solution.  But you can put in a sub-optimal solution, like `lsuc (lsuc (ℓ₁ ⊔ ℓ₂))`.
Except for sizes, Agda only infers unique solutions.

Currently, I do not see an action to take here.  Maybe you can come up with an example where the solution is unique, but Agda does not use it.

**Comment by identicalsnowflake** [comment]
> you can put in a sub-optimal solution.... Except for sizes, Agda only infers unique solutions

I think this is kinda what I'm getting at: I can't think of any time when I've ever intentionally used a sub-optimal level solution, so if the optimal is decidable, even if not unique, I'd be happy to have it use that for `_`.

Alternatively, if widening `_`'s meaning within the language is undesirable, it'd be fine to have `?` and `C-c C-a` do this instead (this currently does not work).

**Comment by nad** [comment]
```agda
record ↑ {a} ℓ (A : Set a) : Set (a ⊔ ℓ) where
  constructor lift
  field lower : A
```

**Comment by andreasabel** [comment]
> Alternatively, if widening _'s meaning within the language is undesirable, it'd be fine to have ? and C-c C-a do this instead (this currently does not work).

I would not mind someone adding such a feature to Agda.

**Comment by identicalsnowflake** [comment]
@nad you're right, it definitely doesn't work everywhere; but it works often enough that I consider it an odd part of my workflow.

**Comment by m0davis** [comment]
@identicalsnowflake, could you write a macro to find the optimal value?

**Comment by identicalsnowflake** [comment]
@m0davis if you had access to the constraint information then probably, but I don't think that the constraints are accessible from the reflection APIs? Reflection gives you access to the goal type, and normally that's enough (in the sense that any value of the goal type will suffice), but Level goals in `Set ?` are weird in that not any value of type `Level` is a valid solution to that `? : Level` hole.

**Comment by gallais** [comment]
May be worth trying the macro approach once #4190 lands.

**Comment by jespercockx** [comment]
There's also some functionality for defaulting levels to their lowest possible value that's used for `--cumulativity`, which could maybe be reused for this feature.
